x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/action/CrossClusterLookupJoinFailuresIT.java

```diff
@@ -0,0 +1,235 @@
+    protected boolean reuseClusters() {
+        return false;
+    }
+
+    public void testLookupFail() throws IOException {
+        setupClusters(3);
+        populateLookupIndex(LOCAL_CLUSTER, "values_lookup", 10);
+        populateLookupIndex(REMOTE_CLUSTER_1, "values_lookup", 25);
+        populateLookupIndex(REMOTE_CLUSTER_2, "values_lookup", 25);
+
+        setSkipUnavailable(REMOTE_CLUSTER_1, true);
+        Exception simulatedFailure = randomFailure();
+        // fail when trying to do the lookup on remote cluster
+        for (TransportService transportService : cluster(REMOTE_CLUSTER_1).getInstances(TransportService.class)) {
+            MockTransportService ts = asInstanceOf(MockTransportService.class, transportService);
+            ts.addRequestHandlingBehavior(
+                EsqlResolveFieldsAction.RESOLVE_REMOTE_TYPE.name(),
+                (handler, request, channel, task) -> handler.messageReceived(request, new TransportChannel() {
+                    @Override
+                    public String getProfileName() {
+                        return channel.getProfileName();
+                    }
+
+                    @Override
+                    public void sendResponse(TransportResponse response) {
+                        sendResponse(simulatedFailure);
+                    }
+
+                    @Override
+                    public void sendResponse(Exception exception) {
+                        channel.sendResponse(exception);
+                    }
+                }, task)
+            );
+        }
+
+        try {
+            // FIXME: this should catch the error but fails instead
+            /*
+            try (
+                EsqlQueryResponse resp = runQuery(
+                    "FROM c*:logs-* | EVAL lookup_key = v | LOOKUP JOIN values_lookup ON lookup_key",
+                    randomBoolean()
+                )
+            ) {
+                var columns = resp.columns().stream().map(ColumnInfoImpl::name).toList();
+                assertThat(columns, hasItems("lookup_key", "lookup_name", "lookup_tag", "v", "tag"));
+
+                List<List<Object>> values = getValuesList(resp);
+                assertThat(values, hasSize(0));
+
+                EsqlExecutionInfo executionInfo = resp.getExecutionInfo();
+
+                var remoteCluster = executionInfo.getCluster(REMOTE_CLUSTER_1);
+                assertThat(remoteCluster.getStatus(), equalTo(EsqlExecutionInfo.Cluster.Status.SKIPPED));
+                assertThat(remoteCluster.getFailures(), not(empty()));
+                var failure = remoteCluster.getFailures().get(0);
+                assertThat(failure.reason(), containsString(simulatedFailure.getMessage()));
+            } */
+
+            try (
+                EsqlQueryResponse resp = runQuery(
+                    "FROM logs-*,c*:logs-* | EVAL lookup_key = v | LOOKUP JOIN values_lookup ON lookup_key",
```

Comment:
nit: it's hard to skim this and see immediately that this query doesn't involve `REMOTE_CLUSTER_2`.

Reply:
Yes, some here only use one cluster, except when the case requires two different remote clusters (such as none next to it). Those are two different cases - one if all remotes miss the index (so the expression matches nothing) and another when one remote misses and one doesn't (so the expression matches something). Those should work the same but it could be different code path so we test both.